### PR TITLE
Add `is.urlInstance`

### LINKED ISSUE #58: Detect `new URL()` instance
`is.url()` might be too confusing though. Maybe `is.urlObject()`? Feedback welcome.

### PR DESCRIPTION
**Pull request by arturovt**
Fixes #58

### CODE REVIEW

**Comment by sindresorhus** [comment]
Don't include commits from a different PR.

**Comment by sindresorhus** [comment]
It needs to be added to the readme.

File: source/index.ts [hunk 1]
```diff
@@ -176,6 +178,7 @@ namespace is { // tslint:disable-line:no-namespace
 	export const dataView = isObjectOfType<DataView>(TypeName.DataView);
 
 	export const directInstanceOf = <T>(instance: any, klass: Class<T>): instance is T => Object.getPrototypeOf(instance) === klass.prototype;
+	export const urlInstance = (value: any): value is URL => !nullOrUndefined(value) && directInstanceOf<URL>(value, URL);
```
Comment (sindresorhus) [inline source/index.ts]: Using instance checking will not work across realms (frames in the browser or `vm` module in Node.js). 

A better solution would be to add `URL` as a type name here: https://github.com/arturovt/is/blob/7e71742a0f6aac5f44be0f2e00ed1619c75fd557/source/index.ts#L53 and then use `isObjectOfType`, like, for example, the `is.generatorFunction` check.

**Comment by sindresorhus** [comment]
Also please include which issue this PR fixes in the PR body in the format of `Fixes #123` (This should always be done on PRs on GitHub).

File: source/index.ts (hunk 1 above)
Comment (arturovt) [reply source/index.ts]: @sindresorhus Agree

File: source/tests/test.ts [hunk 2]
```diff
@@ -533,6 +534,16 @@ test('is.directInstanceOf', t => {
 	t.false(m.directInstanceOf(errorSubclass, Error));
 });
 
+test('is.urlInstance', t => {
+	const url = new URL('https://google.com');
```
Comment (brandon93s) [inline source/tests/test.ts]: http://www.example.com/ is maintained for [documentation](https://www.iana.org/domains/reserved) purposes. I recommend using them in examples, tests and documentation. 

Same for the readme.

**Pull request merged**